### PR TITLE
CUI-7371 [Accessibility] Coral.Datepicker: Use WAI-ARIA 1.2 Combobox design pattern

### PR DESCRIPTION
**Pull request by majornista**
## Description
Refactor Coral.Datepicker to implement WAI-ARIA 1.2 Combobox design pattern, where the `input` has `role="combobox"`, rather than the container element.

This PR also includes a fixes to Calendar for to the use case where no value is specified, so that the Calendar table grid is navigable using the keyboard.

## Related Issue
Per https://jira.corp.adobe.com/browse/CQ-4292426
and https://jira.corp.adobe.com/browse/CUI-7371

## Motivation and Context
Fixes an issue where NVDA get stuck in forms mode and the user is unable to navigate from the input to the dropdown button using the down arrow in browse mode.

## How Has This Been Tested?
Tested with VoiceOver on macOS and NVDA on Windows.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by majornista** [comment]
Please review: @adobre90, @adobe/coral-spectrum, @icaraps, @mohiaror

File: coral-component-datepicker/src/tests/test.Datepicker.js [hunk 1]
```diff
@@ -973,11 +979,12 @@ describe('Datepicker', function() {
         const el = helpers.build(window.__html__['Datepicker.value.html']);
 
         el.on('coral-overlay:open', function() {
-          expect(el.getAttribute('aria-expanded')).to.equal('true');
-        });
+          expect(el._elements.input.getAttribute('aria-expanded')).to.equal('true');
+          expect(el._elements.toggle.getAttribute('aria-expanded')).to.equal('true');        });
```
Comment (icaraps) [inline coral-component-datepicker/src/tests/test.Datepicker.js:983]: missing line break

File: coral-component-calendar/src/scripts/Calendar.js [hunk 2]
```diff
@@ -613,11 +617,8 @@ class Calendar extends BaseFormField(BaseComponent(HTMLElement)) {
     
     if (el) {
       if (isTransitioning) {
-        commons.transitionEnd(newTable, () => {
-          // Prevents from chopping off the transition animation
-          window.setTimeout(() => {
-            el.querySelector('._coral-Calendar-date').classList.add('is-focused');
-          }, 50);
+        window.requestAnimationFrame(() => {
```
Comment (icaraps) [inline coral-component-calendar/src/scripts/Calendar.js:620]: So the animation isn't chopped off neither with `requestAnimationFrame` ?

Comment (majornista) [reply coral-component-calendar/src/scripts/Calendar.js:620]: I'm not noticing animation getting chopped off, although I'm not quite sure what is meant by that, and the movement of focus is much faster without multiple dates getting highlighted when moving quickly between months.

**Comment by majornista** [comment]
Could I get a review: @CezCz, @adobe/coral-spectrum?

**Comment by icaraps** [comment]
Also any particular reason to use coral-dialog in the example ? Would it be possible to use a non coral container in this case ? The example won't work as standalone ie `cd coral-component-datepicker && gulp` because dialog is not imported.

**Comment by majornista** [comment]
@icaraps Trying to ensure that closing the Datepicker using Esc key does not also close the Dialog.

**Comment by majornista** [comment]
@icaraps Please verify. I've removed the dialog example form coral-component-datepicker.

**Pull request merged**